### PR TITLE
Feature 78 overall readiness check

### PR DESCRIPTION
**Pull request by Jdallos**

### CODE REVIEW

File: src/components/tasks/task-dialog.tsx [hunk 1]
```diff
@@ -123,6 +127,8 @@ const TaskDialog = ({ projectId, milestoneId: milestoneIdFromProps, open, task,
     [milestones, taskData.milestoneId],
   );
 
+  console.log("selectedMilestone", selectedMilestone);
```
Comment (lauriLukkarinen) [inline src/components/tasks/task-dialog.tsx:130]:
```suggestion
```

File: src/components/tasks/task-dialog.tsx [hunk 2]
```diff
@@ -828,18 +834,29 @@ const TaskDialog = ({ projectId, milestoneId: milestoneIdFromProps, open, task,
       (selected as string[]).map((id) => (options as Record<string, string>)[id]).join(", ");
   };
 
+  /**
+   *  Checks if milestone project status is planning or initiation
+   */
+  const checkIfMilestonesAreFixed = () => {
+    return project?.status === ProjectStatus.Planning || project?.status === ProjectStatus.Initiation;
+  };
```
Comment (lauriLukkarinen) [inline src/components/tasks/task-dialog.tsx:842]:
```suggestion
  /**
   *  Checks if project status is planning or initiation
   */
  const hasProjectStarted = () => {
    return project?.status !== ProjectStatus.Planning && project?.status !== ProjectStatus.Initiation;
  };
```

**Review by lauriLukkarinen — CHANGES_REQUESTED** [review]

File: src/components/tasks/task-dialog.tsx [hunk 3]
```diff
@@ -828,18 +833,29 @@ const TaskDialog = ({ projectId, milestoneId: milestoneIdFromProps, open, task,
...
+
+    return getValidDateTimeOrThrow(selectedMilestone.startDate);
   };
 
   /**
    * Returns max date for startDate input
    */
   const getStartDateMax = () => {
-    if (!selectedMilestone) return taskData.endDate ?? undefined;
+    if (!selectedMilestone || checkIfMilestonesAreFixed()) return taskData.endDate ?? undefined;
```
Comment (lauriLukkarinen) [inline src/components/tasks/task-dialog.tsx:858]: Is the checkIfMilestonesAreFixed still defined somewhere?

**Review by lauriLukkarinen — CHANGES_REQUESTED** [review]

File: src/components/tasks/task-dialog.tsx (hunk 3 above)
Comment (Jdallos) [reply src/components/tasks/task-dialog.tsx]: The name was changed to hasProjectStarted, I've updated all the references now. Also the logic was flipped from the previous suggestions, so I updated usage in the conditions to be !hasProjectStarted.

**Review by lauriLukkarinen — APPROVED** [review]

**Pull request merged**